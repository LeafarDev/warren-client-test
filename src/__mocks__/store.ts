export default {
  userTotalInboundYield: {
    userTotalInboundYield: {
      total: 0
    }
  },
  transactions: {
    transactions: [],
    currentCreatingTransactionType: 'deposit',
    lastCreatedTransaction: null
  },
  auth: {
    token: ''
  },
  user: {
    user: {
      'id': '1aafa847-b0b4-42c9-929f-61cc75b975a5',
      'name': 'Charles Marciano',
      'email': 'charles.marciano45@gmail.com',
      'accountId': '1539c5cc-8978-41c2-9339-1e34778f5b7d',
      'createdAt': '2021-06-02T01:42:01.987Z',
      'updatedAt': null,
      'account': {
        'id': '1539c5cc-8978-41c2-9339-1e34778f5b7d',
        'accountNumber': '12345678',
        'balance': 0.0056,
        'createdAt': '2021-06-02T01:42:01.766Z',
        'updatedAt': '2021-06-02T21:42:02.128Z'
      }
    }
  },
  isLoading: false
};
