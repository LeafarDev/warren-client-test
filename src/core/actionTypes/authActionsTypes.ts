export const LOGIN = 'authActionTypes/LOGIN';

export interface LoginAction {
  type: typeof LOGIN;
  email: string;
  password: string;
}

export const LOGOUT = 'authActionTypes/LOGOUT';

export interface LogoutAction {
  type: typeof LOGOUT
}

export const RESET = 'authActionTypes/RESET';

export interface ResetAction {
  type: typeof RESET
}

export const LOGIN_REQUEST = 'authActionTypes/LOGIN_REQUEST';

export interface LoginRequestAction {
  type: typeof LOGIN_REQUEST;
}

export const LOGIN_SUCCESS = 'authActionTypes/LOGIN_SUCCESS';

export interface LoginSuccessAction {
  type: typeof LOGIN_SUCCESS;
  token: string;
}

export const LOGIN_FAILURE = 'authActionTypes/LOGIN_FAILURE';

export interface LoginFailureAction {
  type: typeof LOGIN_FAILURE;
  error: Error | string;
}

export type AuthAction =
  | ResetAction
  | LogoutAction
  | LoginRequestAction
  | LoginFailureAction
  | LoginAction
  | LoginSuccessAction;
