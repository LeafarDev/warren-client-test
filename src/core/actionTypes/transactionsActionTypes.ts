import { ITransaction } from '../reducers/transactionsReducer';

export const GET_TRANSACTIONS = 'transactionsActionTypes/GET_TRANSACTIONS';

export interface GetTransactionsAction {
  type: typeof GET_TRANSACTIONS;
}

export const SAVE_TRANSACTION = 'transactionsActionTypes/SAVE_TRANSACTION';

export interface SaveTransactionAction {
  type: typeof SAVE_TRANSACTION;
  transaction: ITransaction
}

export const GET_TRANSACTIONS_REQUEST = 'transactionsActionTypes/GET_TRANSACTIONS_REQUEST';

export interface GetTransactionsRequestAction {
  type: typeof GET_TRANSACTIONS_REQUEST;
}

export const SAVE_TRANSACTION_REQUEST = 'transactionsActionTypes/SAVE_TRANSACTION_REQUEST';

export interface SaveTransactionRequestAction {
  type: typeof SAVE_TRANSACTION_REQUEST;
}

export const GET_TRANSACTIONS_SUCCESS = 'transactionsActionTypes/GET_TRANSACTIONS_SUCCESS';

export interface GetTransactionsSuccessAction {
  type: typeof GET_TRANSACTIONS_SUCCESS;
  transactions: ITransaction[];
}

export const SAVE_TRANSACTION_SUCCESS = 'transactionsActionTypes/SAVE_TRANSACTION_SUCCESS';

export interface SaveTransactionSuccessAction {
  type: typeof SAVE_TRANSACTION_SUCCESS;
  transaction: ITransaction | null;
}

export const CURRENT_CREATING_TRANSACTION_TYPE = 'transactionsActionTypes/CURRENT_CREATING_TRANSACTION_TYPE';

export interface CurrentCreatingTransactionTypeAction {
  type: typeof CURRENT_CREATING_TRANSACTION_TYPE;
  currentCreatingTransactionType: 'withdraw' | 'deposit' | 'payment';
}

export const GET_TRANSACTIONS_FAILURE = 'transactionsActionTypes/GET_TRANSACTIONS_FAILURE';

export interface GetTransactionsFailureAction {
  type: typeof GET_TRANSACTIONS_FAILURE;
  error: Error | string;
}

export const SAVE_TRANSACTION_FAILURE = 'transactionsActionTypes/ SAVE_TRANSACTION_FAILURE';

export interface SaveTransactionFailureAction {
  type: typeof SAVE_TRANSACTION_FAILURE;
  error: Error | string;
}

export type TransactionsAction =
  | CurrentCreatingTransactionTypeAction
  | SaveTransactionAction
  | SaveTransactionRequestAction
  | SaveTransactionSuccessAction
  | SaveTransactionFailureAction
  | GetTransactionsAction
  | GetTransactionsRequestAction
  | GetTransactionsSuccessAction
  | GetTransactionsFailureAction;
