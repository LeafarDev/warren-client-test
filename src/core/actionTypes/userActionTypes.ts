import { IUser } from '../reducers/userReducer';

export const SET_USER = 'userActionTypes/SET_USER';

export interface SetUserAction {
  type: typeof SET_USER;
  user: IUser
}

export const GET_USER = 'userActionTypes/GET_USER';

export interface GetUserAction {
  type: typeof GET_USER;
}

export const GET_USER_REQUEST = 'userActionTypes/GET_USER_REQUEST';

export interface GetUserRequestAction {
  type: typeof GET_USER_REQUEST;
}

export const GET_USER_SUCCESS = 'userActionTypes/GET_USER_SUCCESS';

export interface GetUserSuccessAction {
  type: typeof GET_USER_SUCCESS;
  user: IUser
}

export const GET_USER_FAILURE = 'userActionTypes/GET_USER_FAILURE';

export interface GetUserFailureAction {
  type: typeof GET_USER_FAILURE;
  error: Error | string;
}

export type UserAction =
  | SetUserAction
  | GetUserAction
  | GetUserRequestAction
  | GetUserSuccessAction
  | GetUserFailureAction;
