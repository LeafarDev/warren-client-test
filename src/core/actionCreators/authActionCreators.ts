import * as actions from '../actionTypes/authActionsTypes';

export function login(
  email: string,
  password: string
): actions.LoginAction {
  return {
    type: actions.LOGIN,
    email,
    password
  };
}

export function logout(): actions.LogoutAction {
  return {
    type: actions.LOGOUT
  };
}

export function reset(): actions.ResetAction {
  return {
    type: actions.RESET
  };
}

export function loginRequest(): actions.LoginRequestAction {
  return {
    type: actions.LOGIN_REQUEST
  };
}

export function loginSuccess(
  token: string
): actions.LoginSuccessAction {
  return {
    type: actions.LOGIN_SUCCESS,
    token
  };
}

export function loginFailure(
  error: Error | string
): actions.LoginFailureAction {
  return {
    type: actions.LOGIN_FAILURE,
    error
  };
}
