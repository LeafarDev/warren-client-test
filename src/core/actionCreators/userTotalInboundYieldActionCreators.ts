import * as actions from '../actionTypes/userTotalInboundYieldActionTypes';
import { IUserTotalInboundYield } from '../reducers/userTotalInboundYieldReducer';

export function getUserTotalInboundYield(): actions.GetUserTotalInboundYieldTypesAction {
  return {
    type: actions.GET_USER_TOTAL_INBOUND_YIELD
  };
}

export function getUserTotalInboundYieldRequest(): actions.GetUserTotalInboundYieldTypesRequestAction {
  return {
    type: actions.GET_USER_TOTAL_INBOUND_YIELD_REQUEST
  };
}

export function getUserTotalInboundYieldSuccess(
  userTotalInboundYield: IUserTotalInboundYield
): actions.GetUserTotalInboundYieldTypesSuccessAction {
  return {
    type: actions.GET_USER_TOTAL_INBOUND_YIELD_SUCCESS,
    userTotalInboundYield
  };
}

export function getUserTotalInboundYieldFailure(
  error: Error | string
): actions.GetUserTotalInboundYieldTypesFailureAction {
  return {
    type: actions.GET_USER_TOTAL_INBOUND_YIELD_FAILURE,
    error
  };
}
