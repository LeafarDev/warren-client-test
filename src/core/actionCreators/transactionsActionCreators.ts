import * as actions from '../actionTypes/transactionsActionTypes';
import { ITransaction } from '../reducers/transactionsReducer';

export function getTransactions(): actions.GetTransactionsAction {
  return {
    type: actions.GET_TRANSACTIONS
  };
}

export function saveTransaction(transaction: ITransaction): actions.SaveTransactionAction {
  return {
    type: actions.SAVE_TRANSACTION,
    transaction
  };
}

export function getTransactionsRequest(): actions.GetTransactionsRequestAction {
  return {
    type: actions.GET_TRANSACTIONS_REQUEST
  };
}

export function saveTransactionRequest(): actions.SaveTransactionRequestAction {
  return {
    type: actions.SAVE_TRANSACTION_REQUEST
  };
}

export function getTransactionsSuccess(
  transactions: ITransaction[]
): actions.GetTransactionsSuccessAction {
  return {
    type: actions.GET_TRANSACTIONS_SUCCESS,
    transactions
  };
}

export function saveTransactionSuccess(
  transaction: ITransaction | null
): actions.SaveTransactionSuccessAction {
  return {
    type: actions.SAVE_TRANSACTION_SUCCESS,
    transaction
  };
}

export function currentCreatingTransaction(
  currentCreatingTransactionType: 'withdraw' | 'deposit' | 'payment'
): actions.CurrentCreatingTransactionTypeAction {
  return {
    type: actions.CURRENT_CREATING_TRANSACTION_TYPE,
    currentCreatingTransactionType
  };
}

export function getTransactionsFailure(
  error: Error | string
): actions.GetTransactionsFailureAction {
  return {
    type: actions.GET_TRANSACTIONS_FAILURE,
    error
  };
}

export function saveTransactionFailure(
  error: Error | string
): actions.SaveTransactionFailureAction {
  return {
    type: actions.SAVE_TRANSACTION_FAILURE,
    error
  };
}
