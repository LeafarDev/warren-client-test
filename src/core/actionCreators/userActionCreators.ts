import * as actions from '../actionTypes/userActionTypes';
import { IUser } from '../reducers/userReducer';

export function setUser(user: IUser): actions.SetUserAction {
  return {
    type: actions.SET_USER,
    user
  };
}

export function getUser(): actions.GetUserAction {
  return {
    type: actions.GET_USER
  };
}

export function getUserRequest(): actions.GetUserRequestAction {
  return {
    type: actions.GET_USER_REQUEST
  };
}

export function getUserSuccess(
  user: IUser
): actions.GetUserSuccessAction {
  return {
    type: actions.GET_USER_SUCCESS,
    user
  };
}

export function getUserFailure(
  error: Error | string
): actions.GetUserFailureAction {
  return {
    type: actions.GET_USER_FAILURE,
    error
  };
}
