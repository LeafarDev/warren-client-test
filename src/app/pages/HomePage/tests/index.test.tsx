import React from 'react';
import Index from '..';
import { render, waitFor } from '@testing-library/react';
import * as redux from 'react-redux';
import store from '../../../../__mocks__/store';
import * as userActionCreators from '../../../../core/actionCreators/userActionCreators';
import * as userActionTypes from '../../../../core/actionTypes/userActionTypes';
import * as userTotalInboundYieldActionCreators
  from '../../../../core/actionCreators/userTotalInboundYieldActionCreators';
import * as userTotalInboundYieldActionTypes
  from '../../../../core/actionTypes/userTotalInboundYieldActionTypes';
import * as transactionsActionCreators
  from '../../../../core/actionCreators/transactionsActionCreators';
import * as transactionsActionTypes from '../../../../core/actionTypes/transactionsActionTypes';

jest.mock('react-redux');

describe('<Index />', () => {
  let dispatch;
  beforeEach(() => {
    dispatch = jest.spyOn(redux, 'useDispatch')
      .mockImplementation(jest.fn());
    dispatch.mockReturnValue(jest.fn());
    jest.spyOn(redux, 'useSelector')
      .mockImplementation((cb) => cb(store));

    jest
      .spyOn(userActionCreators, 'getUser')
      .mockImplementation((): userActionTypes.GetUserAction => {
        return {
          type: userActionTypes.GET_USER
        };
      });

    jest
      .spyOn(userTotalInboundYieldActionCreators, 'getUserTotalInboundYield')
      .mockImplementation((): userTotalInboundYieldActionTypes.GetUserTotalInboundYieldTypesAction => {
        return {
          type: userTotalInboundYieldActionTypes.GET_USER_TOTAL_INBOUND_YIELD
        };
      });

    jest
      .spyOn(transactionsActionCreators, 'getTransactions')
      .mockImplementation((): transactionsActionTypes.GetTransactionsAction => {
        return {
          type: transactionsActionTypes.GET_TRANSACTIONS
        };
      });
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(<Index/>);

    expect(firstChild)
      .toMatchSnapshot();
  });

  it('should load user information', async () => {
    render(<Index/>);

    await waitFor(() => {
      expect(userTotalInboundYieldActionCreators.getUserTotalInboundYield)
        .toHaveBeenCalledTimes(1);
      expect(userActionCreators.getUser)
        .toHaveBeenCalledTimes(1);

      expect(transactionsActionCreators.getTransactions)
        .toHaveBeenCalledTimes(1);
    });

  });

});
