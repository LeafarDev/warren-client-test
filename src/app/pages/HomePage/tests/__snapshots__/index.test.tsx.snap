// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> should render and match the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root sc-gtsrHT jtaqrK MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
         
        Welcome Charles Marciano ! Account: 12345678
         
      </h5>
    </div>
    <div
      class="MuiGrid-root sc-bdnxRM jjVypx MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
    >
      <div
        class="MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <div
                class="MuiBox-root MuiBox-root-1"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  Balance
                </p>
                <h2
                  class="MuiTypography-root MuiTypography-h4"
                >
                  $0.00
                </h2>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root sc-bdnxRM jjVypx MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
    >
      <div
        class="MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <div
                class="MuiBox-root MuiBox-root-2"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
                >
                  Total Inbound Yields
                </p>
                <h2
                  class="MuiTypography-root MuiTypography-h4"
                >
                  $0.00
                </h2>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6"
    >
      <div
        class="MuiGrid-root sc-dlnjwi hiknWI MuiGrid-container MuiGrid-align-items-xs-flex-end"
      >
        <div
          class="MuiGrid-root sc-hKFxyN kDxxfH MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-eCApnc iPgxUa"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              New Deposit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root sc-hKFxyN kDxxfH MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-eCApnc iPgxUa"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              New Withdraw
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root sc-hKFxyN kDxxfH MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-eCApnc iPgxUa"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              New Payment
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div>
      <div />
    </div>
    <div
      aria-level="1"
      class="sc-dlMDgC ebNduJ rdt_TableHeader"
      role="heading"
    >
      <div
        class="sc-kfYoZR gZoAjc"
      >
        Transactions
      </div>
      <div
        class="sc-fKgJPI bwRsIf"
      />
      <div
        class="sc-hHEiqL fHFuub"
      >
        <div
          class="sc-iemWCZ hhKMUs"
        />
        <div
          class="sc-dIvrsQ cwqApU"
        />
      </div>
    </div>
    <div
      class="sc-cxNHIi iHCczq"
    >
      <div
        class="sc-iJCRrE jERoqz"
      >
        <div
          class="sc-gKAaRy biSZCX rdt_Table"
          role="table"
        >
          <div
            class="sc-ezzafa icgsDD"
          >
            <div
              style="padding: 24px;"
            >
              There are no records to display
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
